plugins/modules/aws_s3.py

```diff
@@ -155,9 +156,32 @@
     description:
       - KMS key id to use when encrypting objects using I(encrypting=aws:kms). Ignored if I(encryption) is not C(aws:kms).
     type: str
+  copy_src:
+    description:
+    - The source details of the object to copy.
+    - Required if I(mode) is C(copy).
+    type: dict
+    version_added: 2.0.0
+    suboptions:
+      bucket:
+        type: str
+        description:
+        - The name of the source bucket.
+        required: true
+      object:
+        type: str
+        description:
+        - key name of the source object.
+        required: true
+      version_id:
+        type: str
+        description:
+        - version ID of the source object.
+requirements: [ "boto3", "botocore" ]
```

Comment:
Requirements are pulled in from `plugins/doc_fragments/aws.py` with `extends_documentation_fragment` so they don't need to be specified in each module (used in both amazon.aws and community.aws) .